### PR TITLE
[AssertCompareToSpecificMethodRector ] Remove inversion of parameters in Assert calls

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Thanks! I've approved the CI to get feedback. Could you look at it?

To fix this before pushing, run tests locally with `vendor/bin/phpunit`

**Comment by VincentLanglet** [comment]
> Thanks! I've approved the CI to get feedback. Could you look at it?
> 
> To fix this before pushing, run tests locally with `vendor/bin/phpunit`

It should be better now

File: rules-tests/CodeQuality/Rector/MethodCall/AssertCompareToSpecificMethodRector/Fixture/get_class.php.inc [hunk 1]
```diff
@@ -27,7 +27,7 @@ final class GetClass extends \PHPUnit\Framework\TestCase
     public function test()
     {
         $something = new stdClass();
-        $this->assertInstanceOf('stdClass', $something);
+        $this->assertSame(get_class($something), 'stdClass');
```
Comment (TomasVotruba) [inline rules-tests/CodeQuality/Rector/MethodCall/AssertCompareToSpecificMethodRector/Fixture/get_class.php.inc:30]: I think these lines can be removed completely from the tests as unrelated.

File: rules/CodeQuality/Rector/MethodCall/AssertCompareToSpecificMethodRector.php [hunk 2]
```diff
@@ -95,18 +95,12 @@ public function refactor(Node $node): ?Node
...
-        $firstArgumentValue = $firstArgument->value;
         $secondArgumentValue = $secondArgument->value;
 
         if ($secondArgumentValue instanceof FuncCall) {
             return $this->processFuncCallArgumentValue($node, $secondArgumentValue, $firstArgument);
         }
 
-        if ($firstArgumentValue instanceof FuncCall) {
-            return $this->processFuncCallArgumentValue($node, $firstArgumentValue, $secondArgument);
-        }
```
Comment (TomasVotruba) [inline rules/CodeQuality/Rector/MethodCall/AssertCompareToSpecificMethodRector.php:108]: :+1:

File: rules-tests/CodeQuality/Rector/MethodCall/AssertCompareToSpecificMethodRector/Fixture/get_class.php.inc (hunk 1 above)
Comment (VincentLanglet) [reply rules-tests/CodeQuality/Rector/MethodCall/AssertCompareToSpecificMethodRector/Fixture/get_class.php.inc:30]: There is a test for 
```
self::assertSame('stdClass', get_class($something));
```
being fixed by AssertCompareToSpecificMethod

And I would have consider
```
$this->assertSame(get_class($something), 'stdClass');
```
as a non-regression test that if the `get_class` call is made on the first param, nothing change.

In the same way I have added a 
```
$this->assertSame(count($something), 5);
```
as a non-regression test.

Should I remove those two non-regression test ?

Comment (TomasVotruba) [reply rules-tests/CodeQuality/Rector/MethodCall/AssertCompareToSpecificMethodRector/Fixture/get_class.php.inc:30]: Just this line:

>  $this->assertSame(get_class($something), 'stdClass');

File: rules-tests/CodeQuality/Rector/MethodCall/AssertCompareToSpecificMethodRector/Fixture/count.php.inc [hunk 3]
```diff
@@ -25,6 +27,8 @@ final class Count extends \PHPUnit\Framework\TestCase
     public function test()
     {
         $this->assertCount(5, $something);
+        $this->assertSame(count($something), 5);
```
Comment (TomasVotruba) [inline rules-tests/CodeQuality/Rector/MethodCall/AssertCompareToSpecificMethodRector/Fixture/count.php.inc:30]: Could you extract this to a new test fixture? It looks unrelated to change here

Comment (VincentLanglet) [reply rules-tests/CodeQuality/Rector/MethodCall/AssertCompareToSpecificMethodRector/Fixture/count.php.inc]: Done

File: rules-tests/CodeQuality/Rector/MethodCall/AssertCompareToSpecificMethodRector/Fixture/get_class.php.inc (hunk 1 above)
Comment (VincentLanglet) [reply rules-tests/CodeQuality/Rector/MethodCall/AssertCompareToSpecificMethodRector/Fixture/get_class.php.inc]: I moved it in the non-regression-test `skip_first_param`

**Comment by TomasVotruba** [comment]
Perfect! Thank you
